File: tennis-intelligence-platform/api/schemas/match_list.py
"""
schemas/match_list.py — Pydantic response models for the Match Explorer endpoint.

Field names match tennis_intel.serving.match_list_service's own return shape exactly
(see that module's docstring on why "prematch_favourite" compares Elo via
tml_winner_name/tml_loser_name, not MCP's own Player 1/2 slots).
"""

from __future__ import annotations

from pydantic import BaseModel


class MatchSummary(BaseModel):
    match_id: str
    tournament: str
    year: int | None
    surface: str
    round: str
    winner: str
    loser: str
    final_score: str | None
    duration_minutes: int | None
    tournament_level: str | None
    best_of: int | None
    winner_elo: float | None
    loser_elo: float | None
    prematch_favourite: str | None


class MatchListResponse(BaseModel):
    total: int
    limit: int
    offset: int
    matches: list[MatchSummary]

File: tennis-intelligence-platform-web/src/types/matchList.ts
// types/matchList.ts — matches api/schemas/match_list.py exactly.

export interface MatchSummaryRow {
  match_id: string;
  tournament: string;
  year: number | null;
  surface: string;
  round: string;
  winner: string;
  loser: string;
  final_score: string | null;
  duration_minutes: number | null;
  tournament_level: string | null;
  best_of: number | null;
  winner_elo: number | null;
  loser_elo: number | null;
  prematch_favourite: string | null;
}

export interface MatchListResponse {
  total: number;
  limit: number;
  offset: number;
  matches: MatchSummaryRow[];
}

export interface MatchListFilters {
  player?: string;
  tournament?: string;
  year?: number;
  surface?: string;
  round?: string;
  tourney_level?: string;
  best_of?: number;
  limit?: number;
  offset?: number;
}
